tests/phase0/block_processing/test_process_block_header.py

```diff
@@ -14,13 +16,93 @@
 pytestmark = pytest.mark.header
 
 
-def test_proposer_slashed(state):
+def test_sucess(state):
+    pre_state = deepcopy(state)
+    block = build_empty_block_for_next_slot(pre_state)
+
+    #
+    # setup pre_state to be ready for block transition
+    #
+    cache_state(pre_state)
+    advance_slot(pre_state)
+
+    post_state = deepcopy(pre_state)
+
+    #
+    # test block header
+    #
+    process_block_header(post_state, block)
+
+    return state, [block], post_state
+
+
+def test_invalid_slot(state):
+    pre_state = deepcopy(state)
+
+    # mess up previous block root
+    block = build_empty_block_for_next_slot(pre_state)
+    block.previous_block_root = b'\12'*32
```

Comment:
It seems should only be set in the `test_invalid_previous_block_root` case?

Reply:
`build_empty_block_for_next_slot` builds a valid block including the previous_block_root. Just overriding in the case where we want to trigger an error.

Reply:
But the only difference between `test_invalid_slot` and `test_invalid_previous_block_root` is that `test_invalid_slot` has one more `advance_slot(pre_state)`; and they both throw `AssertionError`. So we are not able to see what's the root cause that triggers the exception in `test_invalid_slot` case.

Reply:
I missed the error you were actually pointing out. Fixed!